fastapi_azure_auth/auth.py

```diff
@@ -180,48 +194,41 @@ async def __call__(self, request: Request, security_scopes: SecurityScopes) -> O
             try:
                 if key := self.openid_config.signing_keys.get(header.get('kid', '')):
                     # We require and validate all fields in an Azure AD token
+                    required_claims = ['exp', 'aud', 'iat', 'nbf', 'sub']
+                    if self.validate_iss:
+                        required_claims.append('iss')
+
                     options = {
                         'verify_signature': True,
                         'verify_aud': True,
                         'verify_iat': True,
                         'verify_exp': True,
                         'verify_nbf': True,
                         'verify_iss': self.validate_iss,
-                        'verify_sub': True,
-                        'verify_jti': True,
-                        'verify_at_hash': True,
-                        'require_aud': True,
-                        'require_iat': True,
-                        'require_exp': True,
-                        'require_nbf': True,
-                        'require_iss': self.validate_iss,
-                        'require_sub': True,
-                        'require_jti': False,
-                        'require_at_hash': False,
-                        'leeway': self.leeway,
+                        'require': required_claims,
                     }
                     # Validate token
-                    token = jwt.decode(
-                        access_token,
-                        key=key,
-                        algorithms=['RS256'],
-                        audience=self.app_client_id if self.token_version == 2 else f'api://{self.app_client_id}',
-                        issuer=iss,
-                        options=options,
-                    )
+                    token = self.validate(access_token=access_token, iss=iss, key=key, options=options)
                     # Attach the user to the request. Can be accessed through `request.state.user`
                     user: User = User(
                         **{**token, 'claims': token, 'access_token': access_token, 'is_guest': user_is_guest}
                     )
                     request.state.user = user
                     return user
-            except JWTClaimsError as error:
+            except (
+                InvalidAudienceError,
+                InvalidIssuerError,
+                InvalidIssuedAtError,
+                ImmatureSignatureError,
```

Comment:
Signature is probably not a claim error, right?

Reply:
From what I can see, an invalid `nbf` will raise `ImmatureSignatureError`. So, it seems like a claim error, following the current logic. There is also `InvalidSignatureError` but that will fall on the `InvalidTokenError` except.

Reply:
Ah, I see. 

Thanks for the clarifications. 

I'll merge and release tomorrow! Thanks again 😊

Reply:
Thanks, that will be great, glad I can help.